src/machine/machine_avr.go

```diff
@@ -157,3 +157,96 @@ func (a ADC) Get() uint16 {
 	high := uint16(*avr.ADCH)
 	return uint16(low) | uint16(high<<8)
 }
+
+// I2C on the Arduino
+
+// I2CInit initializes the I2C interface.
+func I2CInit() {
+	// Activate internal pullups for twi.
+	*avr.PORTC |= (avr.DIDR0_ADC4D | avr.DIDR0_ADC5D)
+
+	// Initialize twi prescaler and bit rate.
+	*avr.TWSR |= (avr.TWSR_TWPS0 | avr.TWSR_TWPS1)
+
+	// twi bit rate formula from atmega128 manual pg. 204:
+	// SCL Frequency = CPU Clock Frequency / (16 + (2 * TWBR))
+	// NOTE: TWBR should be 10 or higher for master mode.
+	// It is 72 for a 16mhz board with 100kHz TWI
+	*avr.TWBR = avr.RegValue(72)
```

Comment:
You can calculate this value at compile time with constants, to avoid hardcoding this value in the I2C layer.
For example, you can add a `CPU_FREQUENCY` (or something like that) to machine_arduino.go with value `16000000`, and use it here to calculate the TWBR frequency.